parentchain/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use crate::weights::WeightInfo;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// Configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type WeightInfo: WeightInfo;
+	}
+
+	/// The current block number being processed. Set by `execute_block`.
+	#[pallet::storage]
+	#[pallet::getter(fn block_number)]
+	pub(super) type Number<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+	/// Hash of the previous block.
+	#[pallet::storage]
+	#[pallet::getter(fn parent_hash)]
+	pub(super) type ParentHash<T: Config> = StorageValue<_, T::Hash, ValueQuery>;
+
+	/// Map of block numbers to block hashes.
+	#[pallet::storage]
+	#[pallet::getter(fn block_hash)]
+	pub(super) type BlockHash<T: Config> = StorageValue<_, T::Hash, ValueQuery>;
```

Comment:
The comment and the storage do not match. What you have here is not a Map from BlockNumber -> BlockHash but just the storage of the last BlockHash. For a Map you need to set the BlockNumber as a key.

Example of a storage map: https://github.com/integritee-network/pallets/blob/master/teeracle/src/lib.rs#L55-L57

Reply:
Unresolving this conversation, because it is not yet clear for me why you've decided to go for the last block hash instead of a map from `BlockNumber` -> `BlockHash.` Is there a specific reason?

Reply:
Sorry I misunderstood. Why do we need a map? Why the simple storage is not enough?

Reply:
I don't think we'd want an ever-growing map in the pallet, this bloats the runtime storage indefinitely. If such a continuous mapping is wished for, this should be handled outside the runtime.

Reply:
Hm. The frame pallet does include this mapping. But agreed, it might be an overkill. Let's leave it as it  is because its only a template anyway.

Reply:
🎉